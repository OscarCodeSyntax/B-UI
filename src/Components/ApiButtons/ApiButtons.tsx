import * as React from "react";
import BasicQueries from "../../Resources/Queries/BasicQueries";
import { useState } from "react";
import Button from "@mui/material/Button";
// import "./App.css";
import "./../../App.css";

const ApiButtons = () => {

  const [screenValue, setScreenValue] = useState<string>("Null");

  const deleteOnClick = async () => {
    if (await BasicQueries.deleteById(2)) {
      setScreenValue("Deleted");
    }
  };

  const findAllOnClick = async () => {
    if (await BasicQueries.findAll()) {
      setScreenValue("Found All");
    }
  };

  const updateOnClick = async () => {
    if (await BasicQueries.update(2, "updated")) {
      setScreenValue("Updated");
    }
  };

  return (
    <div className="width-250px">
      <div className="text-center"> {screenValue}</div>
      <div className="padding-10">
        <Button
          className="width-250px"
          variant="outlined"
          onClick={(e: any) =>
            BasicQueries.create("create").then(setScreenValue)
          }
          title="Create"
          value="Create"
        >
          Create
        </Button>
      </div>
      <div className="padding-10 ">
        <Button
          className="width-250px"
          variant="outlined"
          onClick={(e: any) => {
            deleteOnClick();
          }}
          title="Delete By Id"
          value="Delete By Id"
        >
          Delete By Id
        </Button>
      </div>
      <div className="padding-10">
        <Button
          className="width-250px"
          variant="outlined"
          onClick={(e: any) => {
            findAllOnClick();
          }}
          title="Find All"
          value="Find All"
        >
          Find All
        </Button>
      </div>
      <div className="padding-10">
        <Button
          className="width-250px"
          variant="outlined"
          onClick={(e: any) => {
            BasicQueries.findById(1).then(setScreenValue);
          }}
          title="Find By Id"
          value="Find By Id"
        >
          Find By Id
        </Button>
      </div>
      <div className="padding-10">
        <Button
          className="width-250px"
          variant="outlined"
          onClick={(e: any) => {
            updateOnClick();
          }}
          title="Update"
          value="Update"
        >
          Update
        </Button>
      </div>
    </div>
  );
};

export default ApiButtons;
