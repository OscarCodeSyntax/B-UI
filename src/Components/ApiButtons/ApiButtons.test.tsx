import React from "react";
import ApiButtons from "./ApiButtons";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";

describe("Api Buttons ", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly", () => {
    const ApiButtonsWrapper = renderer.create(<ApiButtons />).toJSON();
    expect(ApiButtonsWrapper).toMatchSnapshot();
  });

  it("renders 5 buttons elements", () => {
    const testRenderer = renderer.create(<ApiButtons />);
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("button").length).toEqual(5);
  });

  it("has all the correct text", () => {
    render(<ApiButtons />);
    expect(screen.getByText("Create")).toBeTruthy();
    expect(screen.getByText("Delete By Id")).toBeTruthy();
    expect(screen.getByText("Update")).toBeTruthy();
    expect(screen.getByText("Find By Id")).toBeTruthy();
    expect(screen.getByText("Find All")).toBeTruthy();
  });

  it("display value should change after each click", () => {
    render(<ApiButtons />);
    expect(screen.getByText("Null")).toBeTruthy();

    fireEvent.click(screen.queryByText("Create"));
    expect(screen.findByText('create reached : "create"')).toBeTruthy();

    fireEvent.click(screen.queryByText("Delete By Id"));
    expect(screen.findByText("Deleted")).toBeTruthy();

    fireEvent.click(screen.queryByText("Find All"));
    expect(screen.findByText("Found All")).toBeTruthy();

    fireEvent.click(screen.queryByText("Find By Id"));
    expect(screen.findByText("get by Id reached : 1")).toBeTruthy();

    fireEvent.click(screen.queryByText("Update"));
    expect(screen.findByText("Updated")).toBeTruthy();
  });
});
