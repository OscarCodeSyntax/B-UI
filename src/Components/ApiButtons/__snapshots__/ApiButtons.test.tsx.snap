// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api Buttons  renders correctly 1`] = `
<div
  className="width-250px"
>
  <div
    className="text-center"
  >
     
    Null
  </div>
  <div
    className="padding-10"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium width-250px css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Create"
      type="button"
      value="Create"
    >
      Create
    </button>
  </div>
  <div
    className="padding-10 "
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium width-250px css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Delete By Id"
      type="button"
      value="Delete By Id"
    >
      Delete By Id
    </button>
  </div>
  <div
    className="padding-10"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium width-250px css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Find All"
      type="button"
      value="Find All"
    >
      Find All
    </button>
  </div>
  <div
    className="padding-10"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium width-250px css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Find By Id"
      type="button"
      value="Find By Id"
    >
      Find By Id
    </button>
  </div>
  <div
    className="padding-10"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium width-250px css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Update"
      type="button"
      value="Update"
    >
      Update
    </button>
  </div>
</div>
`;
